### PR TITLE
[WIP] Move /ok-to-test to be handled in GenericCommentEvent

### LINKED ISSUE #3827: ok-to-test not detected in review approval
Example: https://github.com/kubernetes/kubernetes/pull/49872

Other commands like `/lgtm` are

### PR DESCRIPTION
**Pull request by cblecker**
Fixes #3827.

I think this will work, but it does muddy things with labels a bit. When you pull down a PR as an "Issue" then you get things like the labels applied. That's not the case for a PR object however.

### CODE REVIEW

File: prow/plugins/trigger/ic_test.go [hunk 1]
```diff
@@ -166,33 +166,6 @@ func TestHandleIssueComment(t *testing.T) {
 			ShouldBuild:   true,
 			StartsExactly: "pull-jib",
 		},
-		{
-			name: "needs-ok-to-test label is removed when no presubmit runs by default",
```
Comment (kargakis) [inline prow/plugins/trigger/ic_test.go]: Why is this test removed?

**Comment by cjwagner** [comment]
We have avoided handling `/ok-to-test` with generic comment events because the trigger plugin's `trustedPullRequest` function has to look for the presence of an `/ok-to-test` command and if we accepted generic comment events instead of just issue comment events we would need to list issue comments, pull request reviews, and pull request review comments instead of just issue comments. Since trigger is a heavily used plugin tripling the token usage to support handling `/ok-to-test` in reviews and review comments might not be worth it. @kargakis WDYT?

File: prow/plugins/trigger/ic_test.go (hunk 1 above)
Comment (kargakis) [reply prow/plugins/trigger/ic_test.go]: Now I see, nvmd

**Comment by kargakis** [comment]
> We have avoided handling /ok-to-test with generic comment events because the trigger plugin's trustedPullRequest function has to look for the presence of an /ok-to-test command and if we accepted generic comment events instead of just issue comment events we would need to list issue comments, pull request reviews, and pull request review comments instead of just issue comments. Since trigger is a heavily used plugin tripling the token usage to support handling /ok-to-test in reviews and review comments might not be worth it. @kargakis WDYT?

It might be worth it to handle it for all cases since people are already trying to use it in reviews but I can't say for sure w/o the ability to figure out the percentage of tokens each plugin consumes every hour (until refreshed).

**Comment by cblecker** [comment]
Closing this stale PR. Still to be addressed in https://github.com/kubernetes/test-infra/issues/3827

**Pull request closed without merging**